app/services/redis_session_health_checker.rb

```diff
@@ -0,0 +1,30 @@
+module RedisSessionHealthChecker
+  module_function
+
+  # @return [HealthCheckSummary]
+  def check
+    HealthCheckSummary.new(healthy: true, result: health_write_and_read)
```

Comment:
Similar to the Outbound health check, would we want to cache this for a period of time so that we mitigate this as a potential vector for Denial of Service? I suppose we'd want to be careful to not use Redis as the cache as well.

Reply:
We are multiproc and multithread so getting a per-instance global might be tricky.   (*For me, a Ruby n00b.)

That said we are talking hundreds of calls a minute for normal health checking.  Some sort of caching is a good idea.

Reply:
ooo when did we upgrade to multithreaded?

Reply:
Haven't yet @zachmargolis...   I updated my note.

Reply:
> That said we are talking hundreds of calls a minute for normal health checking. Some sort of caching is a good idea.

Maybe a whole separate convo, but if this health check is happening so frequently, should we break redis health into a separate endpoint? Like separate high frequency "can you serve traffic" health check endpoints from medium frequency "are all connections working 100% normal"?

Because if redis is down, typically it's down for all instances, so doing one quick check across all instances every so often (1/minute) is fine vs 1/instance/minute or whatever?

Reply:
I am thinking of some nasty failure modes like if Redis is unavailable from one of the AZs - We want all the instances in that AZ to go bye-bye.   A Redis call is cheap and made for nearly every page served, which makes me worry about the frequency this will run less.

Reply:
We spoke a bit and I think we should defer caching unless it becomes a problem.